soccer/Processor.cpp

```diff
@@ -590,6 +602,74 @@ void Processor::run() {
     vision.stop();
 }
 
+/*
+ * Updates the geometry packet if different from the existing one,
+ * Based on the geometry vision data.
+ */
+void Processor::updateGeometryPacket(const SSL_GeometryFieldSize* fieldSize) {
+    if (fieldSize->field_lines_size() == 0) {
+        return;
+    }
+
+    const SSL_FieldCicularArc* penalty = nullptr;
+    const SSL_FieldCicularArc* center = nullptr;
+    float displacement =
+        Field_Dimensions::Default_Dimensions.GoalFlat();  // default displacment
+
+    for (int i = 0; i < fieldSize->field_arcs().size(); i++) {
```

Comment:
Note that an easier way to do loops over elements in a collection is with this syntax:

~~~{.cpp}
for (const SSL_FieldCircularArc& arc : fieldSize->field_arcs()) {

}
~~~